modules/schematics/src/create-action-migration/index.ts

```diff
@@ -0,0 +1,206 @@
+              actionClass.getEnd()
+            ),
+            new InsertChange(
+              path,
+              typeVariable[0].getStart(sourceFile),
+              createActionDefinition
+            ),
+          ];
+        })
+        .filter(Boolean) as []).reduce(
+        (imports, curr) => imports.concat(curr),
+        []
+      );
+
+      let importChanges: (ReplaceChange | RemoveChange | Change)[] = [];
+      const imports = [];
+      if (changes.length > 0) {
+        imports.push('createAction');
+        if (importProps) {
+          imports.push('props');
+        }
+        importChanges = replaceImport(
+          sourceFile,
+          path,
+          '@ngrx/store',
+          'Action',
+          imports.join(', ')
+        );
+      }
+
+      return commitChanges(host, sourceFile.fileName, [
+        ...changes,
+        ...importChanges,
+      ]);
+    });
+}
+
+const isActionImported = (sourceFile: ts.SourceFile): boolean => {
+  return !!sourceFile.statements
+    .filter(ts.isImportDeclaration)
+    .filter(({ moduleSpecifier }) =>
+      moduleSpecifier.getText(sourceFile).includes('@ngrx/store')
+    )
+    .filter(p => {
+      return (p.importClause!
+        .namedBindings! as ts.NamedImports).elements.filter(spec => {
+        if (spec.name.text) {
+          return spec.name.text === 'Action';
+        }
+      }).length;
+    }).length;
+};
+
+const getActionClasses = (sourceFile: ts.SourceFile): ts.ClassDeclaration[] => {
+  return sourceFile.statements
+    .filter(ts.isClassDeclaration)
+    .filter(
+      statement => statement.heritageClauses && statement.heritageClauses.length
+    )
+    .filter(statement => {
+      return statement.heritageClauses!.filter(clause => {
+        return (clause.types || []).filter(type => {
+          return (
+            type.expression && type.expression.getText(sourceFile) === 'Action'
+          );
+        }).length;
+      }).length;
+    });
+};
+
+const getConstructorPayloadType = (
+  host: Tree,
+  path: Path,
+  actionClass: ts.ClassDeclaration
+): string => {
+  let payloadType = '';
+
+  let payloadInConstructor = actionClass.members
+    .filter(ts.isConstructorDeclaration)
+    .filter(member => member.parameters.filter(param => param.type).length)
+    .map(member => member.parameters[0].type)[0];
+
+  if (payloadInConstructor) {
+    payloadType = host
+      .read(path)!
+      .toString('utf8')
+      .substring(payloadInConstructor.pos, payloadInConstructor.end)
+      .trim();
+  }
+
+  return payloadType;
+};
+
+const composeCreateAction = (
+  typeValue: string,
+  payloadType: string,
+  actionClass: ts.ClassDeclaration
+): string => {
+  let actionName = camelize(actionClass.name!.text.replace('Action', ''));
+  let createAction = tags.stripIndent`
+        export const ${actionName} = createAction(
+          '${typeValue}'`;
+  if (payloadType) {
+    createAction += `,
+  props<{ payload: ${payloadType} }>()`;
```

Comment:
I think this is wrong, this would mean that even if the action doesn't have a `payload` but a `msg` for example, we would rename it to `payload`.

We should only migrate the ones with `payload`, or we should keep the naming (and the number of props).
I would prefer the latter.